src/server/footballContent.ts

```diff
@@ -1,66 +1,157 @@
 
-const getFootballSelector = (
-	date: Option<Date>,
-	teamA: Option<string>,
-	teamB: Option<string>,
-): Option<string> => {
-	if (
-		date.kind === OptionKind.Some &&
-		teamA.kind === OptionKind.Some &&
-		teamB.kind === OptionKind.Some
-	) {
-		// MAPI sorts these by string value
-		const teams =
-			teamA.value < teamB.value
-				? [teamA.value, teamB.value]
-				: [teamB.value, teamA.value];
-
-		const d = date.value;
-		const year = d.getUTCFullYear();
-		const month = padZero(d.getUTCMonth() + 1);
-		const day = padZero(d.getUTCDate());
-
-		return some(`${year}-${month}-${day}_${teams[0]}_${teams[1]}`);
-	}
+const getFootballSelector = (date: Date, [teamA, teamB]: Teams): string => {
+	// MAPI sorts these by string value
+	const teams = teamA < teamB ? [teamA, teamB] : [teamB, teamA];
 
-	return none;
-};
+	const d = date;
+	const year = d.getUTCFullYear();
+	const month = padZero(d.getUTCMonth() + 1);
+	const day = padZero(d.getUTCDate());
 
-const getFootballEndpoint = (selectorId: Option<string>): Option<string> => {
-	if (selectorId.kind === OptionKind.Some) {
-		return some(
-			`https://mobile.guardianapis.com/sport/football/matches?selector=${selectorId.value}`,
-		);
-	}
-	return none;
+	return `${year}-${month}-${day}_${teams[0]}_${teams[1]}`;
 };
 
-type FootballResponse = Record<string, FootballContent>;
+const getFootballEndpoint = (selectorId: string): string =>
+	`https://mobile.guardianapis.com/sport/football/matches?selector=${selectorId}`;
+
+const optionToUndefined = <A>(optA: Option<A>): A | undefined =>
+	withDefault<A | undefined>(undefined)(optA);
+
+const stringOrUndefinedParser: Parser<string | undefined> = pipe(
+	stringParser,
+	maybe,
+	map(optionToUndefined),
+);
+
+const makeScorer = (
+	player: string,
+	timeInMinutes: number,
+	additionalInfo: undefined | string,
+): Scorer => ({
+	player,
+	timeInMinutes,
+	additionalInfo,
+});
+
+const makeFootballTeam = (
+	id: string,
+	name: string,
+	shortCode: string,
+	crestUri: string,
+	score: number,
+	scorers: Scorer[],
+): FootballTeam => ({
+	id,
+	name,
+	shortCode,
+	crestUri,
+	score,
+	scorers,
+});
+
+const makeFootballContent = (
+	id: string,
+	status: string,
+	kickOff: string,
+	competitionDisplayName: string,
+	homeTeam: FootballTeam,
+	awayTeam: FootballTeam,
+	venue: string | undefined,
+): FootballContent => ({
+	id,
+	status,
+	kickOff,
+	competitionDisplayName,
+	homeTeam,
+	awayTeam,
+	venue,
+});
+
+const scorerParser: Parser<Scorer> = map3(makeScorer)(
+	fieldParser('player', stringParser),
+	fieldParser('timeInMinutes', numberParser),
+	fieldParser('additionalInfo', stringOrUndefinedParser),
+);
+
+const footballTeamParser: Parser<FootballTeam> = map6(makeFootballTeam)(
+	fieldParser('id', stringParser),
+	fieldParser('name', stringParser),
+	fieldParser('shortCode', stringParser),
+	fieldParser('crestUri', stringParser),
+	fieldParser('score', numberParser),
+	fieldParser('scorers', arrayParser(scorerParser)),
```

Comment:
To handle the case where `scorers` is missing you could do this:

```suggestion
	oneOf([
		fieldParser('scorers', arrayParser(scorerParser)),
		succeed([]),
	]),
```